engine/src/operators/GroupBy.cpp

```diff
@@ -214,28 +213,44 @@ std::unique_ptr<ral::frame::BlazingTable> compute_groupby_without_aggregations(
 }
 
 std::unique_ptr<ral::frame::BlazingTable> compute_aggregations_without_groupby(
-		const ral::frame::BlazingTableView & table, const std::vector<std::string> & aggregation_input_expressions,
-		const std::vector<AggregateKind> & aggregation_types, const std::vector<std::string> & aggregation_column_assigned_aliases){
-
+	const ral::frame::BlazingTableView & table,
+	const std::vector<std::string> & aggregation_input_expressions,
+	const std::vector<AggregateKind> & aggregation_types,
+	const std::vector<std::string> & aggregation_column_assigned_aliases)
+{
 	std::vector<std::unique_ptr<cudf::scalar>> reductions;
 	std::vector<std::string> agg_output_column_names;
+	std::vector<int> all_indices;
+	bool is_multi_var = false;
 	for (size_t i = 0; i < aggregation_types.size(); i++){
-		if(aggregation_input_expressions[i] == "" && aggregation_types[i] == AggregateKind::COUNT_ALL) { // this is a COUNT(*)
+		std::string agg_input_expr = aggregation_input_expressions[i];
+		if (agg_input_expr == "" && aggregation_types[i] == AggregateKind::COUNT_ALL) { // this is a COUNT(*)
 			std::unique_ptr<cudf::scalar> scalar = cudf::make_numeric_scalar(cudf::data_type(cudf::type_id::INT64));
 			auto numeric_s = static_cast< cudf::scalar_type_t<int64_t>* >(scalar.get());
 			numeric_s->set_value((int64_t)(table.view().num_rows()));
 			reductions.emplace_back(std::move(scalar));
 		} else {
 			std::vector<std::unique_ptr<ral::frame::BlazingColumn>> aggregation_input_scope_holder;
-			CudfColumnView aggregation_input;
-			if(is_var_column(aggregation_input_expressions[i]) || is_number(aggregation_input_expressions[i])) {
-				aggregation_input = table.view().column(get_index(aggregation_input_expressions[i]));
+			cudf::column_view aggregation_input;
+
+			// For operations like `COUNT($0, $1)`
+			size_t num_columns_to_count = StringUtil::findAndCountAllMatches(agg_input_expr, "$");
+			if (num_columns_to_count > 1 && aggregation_types[i] == AggregateKind::COUNT_VALID) {
+				is_multi_var = true;
+				all_indices = get_all_indices_from_count_expression(agg_input_expr);
```

Comment:
I think you should combine `get_all_indices_from_count_expression` and `expand_indices_to_apply_casts` into one function. They are only ever used here, so it would make more sense to combine them and call the function something very explicit to this very purpose. Something like:
`modify_multi_column_count_expression`

Reply:
ok, got it.